### PR TITLE
Deployer tests and Debt token borrower/lender roles.

### PR DESCRIPTION
**Pull request by kosecki123**
DebtTokenDeployer tests. 
DebtToken explicitly requires borrower and lender to be set.

### CODE REVIEW

**Comment by kosecki123** [comment]
```
function fetchDayTokens() onlyOwner public {
        dayToken.transfer(owner, dayToken.balanceOf(this));
    }
```

was not covered by tests. I think we shall get rid of this function and send the fee tokens directly to owner

instead of `if(dayToken.transferFrom(user, this, dayTokenFees)){`
use `if(dayToken.transferFrom(user, owner, dayTokenFees)){`

File: contracts/DebtToken.sol [hunk 1]
```diff
@@ -74,6 +76,16 @@ contract DebtToken is Ownable {
     return balances[_owner];
   }
 
+  modifier onlyBorrower() {
+    require(isBorrower());
+    _;
+  }
+
+  modifier onlyLender() {
+    require(isBorrower());
```
Comment (adibas03) [inline contracts/DebtToken.sol]: This should be
`isLender()`

File: contracts/DebtToken.sol [hunk 2]
```diff
@@ -106,7 +117,7 @@ contract DebtToken is Ownable {
    * @dev Function to stop minting new tokens.
    * @return True if the operation was successful.
    */
-  function finishMinting() onlyOwner private returns (bool) {
+  function finishMinting() onlyBorrower private returns (bool) {
```
Comment (adibas03) [inline contracts/DebtToken.sol]: `private` should be `internal`.
We do not want it called except by this contract.
Private allows any contract that uses this contract call it.

File: contracts/DebtToken.sol [hunk 3]
```diff
@@ -184,7 +195,7 @@ contract DebtToken is Ownable {
   calculate the total number of passed interest cycles and coin value
   */
   function calculateInterestDue() public constant returns(uint256 _coins,uint256 _cycle){
-    if(!isTermOver())
+    if(!isTermOver() || !isLoanFunded())
```
Comment (adibas03) [inline contracts/DebtToken.sol]: ` || !isLoanFunded()` is already implicit in `isTermOver()`

File: test/debtToken.js [hunk 4]
```diff
@@ -245,7 +243,7 @@ contract('DebtToken', function(accounts){
         });
         
         it('Should confirm loanValue does not increase after refundLoan',function(done){
-          var time = deployment_config._loanCycle*2*deployment_config._dayLength*1000;
+          var time = deployment_config._loanCycle*2*deployment_config._dayLength*2000;
```
Comment (adibas03) [inline test/debtToken.js]: could you explain the increase to `2000` from `1000`?

File: contracts/DebtToken.sol (hunk 1 above)
Comment (kosecki123) [reply contracts/DebtToken.sol]: That is correct

File: contracts/DebtToken.sol (hunk 2 above)
Comment (kosecki123) [reply contracts/DebtToken.sol]: `private` means that only current contract can call it, `internal` allows the contract that inherits from the contact.

We indeed can change it to `internal` shall be better option for the future and building on top of debt contract.

File: contracts/DebtToken.sol (hunk 3 above)
Comment (kosecki123) [reply contracts/DebtToken.sol]: This isn't actually redundant. Removing it causes `Should confirm loanValue does not increase after refundLoan` test to fail.

File: test/debtToken.js (hunk 4 above)
Comment (kosecki123) [reply test/debtToken.js]: To be reverted, some testing leftovers.

File: contracts/DebtToken.sol (hunk 3 above)
Comment (adibas03) [reply contracts/DebtToken.sol]: will check that out

File: contracts/DebtTokenDeployer.sol [hunk 5]
```diff
@@ -33,14 +32,16 @@ contract DeployDebtToken is Ownable{
         uint256 _loanTerm,
         uint256 _loanCycle,
         uint256 _intrestRatePerCycle,
-        address _lender){
+        address _lender)
+    public
+    {
 
         address user = msg.sender;
```
Comment (adibas03) [inline contracts/DebtTokenDeployer.sol]: There is no need to create a new variable
just use `msg.sender` on lines L#41 and L#44
Additional variable only increases gas cost, nothing else

File: contracts/DebtToken.sol [hunk 6]
```diff
@@ -42,12 +42,23 @@ contract DebtToken is Ownable {
...
       uint256 _interestRatePerCycle,
-      address _lender
+      address _lender,
+      address _borrower
       ) {
+
+      require(_exchangeRate > 0);
+      require(_initialAmount > 0);
+      require(_dayLength > 0);
+      require(_loanTerm > 0);
```
[the remark this answers is not in the document]
Comment (kosecki123) [reply contracts/DebtToken.sol]: Agree

**Pull request merged**